dominion/src/main/java/org/codice/dominion/options/Options.java

```diff
@@ -0,0 +1,621 @@
+/**
+ * Copyright (c) Codice Foundation
+ *
+ * <p>This is free software: you can redistribute it and/or modify it under the terms of the GNU
+ * Lesser General Public License as published by the Free Software Foundation, either version 3 of
+ * the License, or any later version.
+ *
+ * <p>This program is distributed in the hope that it will be useful, but WITHOUT ANY WARRANTY;
+ * without even the implied warranty of MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+ * GNU Lesser General Public License for more details. A copy of the GNU Lesser General Public
+ * License is distributed along with this program and can be found at
+ * <http://www.gnu.org/licenses/lgpl.html>.
+ */
+package org.codice.dominion.options;
+
+import java.lang.annotation.Documented;
+import java.lang.annotation.ElementType;
+import java.lang.annotation.Inherited;
+import java.lang.annotation.Repeatable;
+import java.lang.annotation.Retention;
+import java.lang.annotation.RetentionPolicy;
+import java.lang.annotation.Target;
+import java.util.concurrent.TimeUnit;
+import org.codice.dominion.conditions.Conditions;
+import org.codice.dominion.interpolate.Interpolate;
+import org.codice.dominion.options.Option.Annotation;
+import org.codice.dominion.options.Options.Repeatables.UpdateConfigFiles;
+
+/**
+ * This class defines annotations that can be used to configure containers. It is solely used for
+ * scoping.
+ */
+public class Options {
+  /**
+   * Constant internally used in annotation definition to indicate a particular <code>String</code>
+   * attribute is not yet defined.
+   */
+  public static final String NOT_DEFINED = "_not_defined_";
+
+  /** Annotation used to specify a maven URL reference. */
+  @Retention(RetentionPolicy.RUNTIME)
+  @Inherited
+  @Documented
+  public @interface MavenUrl {
+    /**
+     * Constant to use when the actual version, type, and/or classifier of the referenced artifact
+     * should be retrieved from the maven project's <code>dependencies.properties</code> file where
+     * the annotation is used. It does require the project to use the depends-maven-plugin plugin in
+     * order for the file to be generated and added to the jar/bundle artifact.
+     */
+    public static final String AS_IN_PROJECT = "_as_in_project_";
```

Comment:
They serve 2 different needs. `AS_IN_PROJECT` returns a maven properties based on a given artifact. For example, you have a dependency to commons-io but need to get the version "as" defined "in" the "project". `AS_PROJECT` is used to retrieve attributes of the current project such as its group id, artifact id, or version. This could be useful in case where the current project is also installing a `features.xml` in addition to a jar/bundle and you which to avoid hardcoding in 2 places (i.e. the pom and the annotation) this information.